new-components/reporters/elasticsearch/internal/reporter/reporter.go

```diff
@@ -0,0 +1,155 @@
+
+	"github.com/smithy-security/pkg/env"
+
+	vf "github.com/smithy-security/smithy/sdk/component/vulnerability-finding"
+
+	"github.com/smithy-security/smithy/sdk/component"
+)
+
+type Conf struct {
+	ElasticsearchURL    string
+	ElasticsearchIndex  string
+	ElasticsearchApiKey string
+}
+
+type esLogger struct {
+	Conf *Conf
+}
+
+// NewJsonLogger returns a new json logger.
+func NewElasticsearchLogger(c *Conf) esLogger {
+	return esLogger{
+		Conf: c,
+	}
+}
+
+// NewConf returns a new configuration build from environment lookup.
+func NewConf(envLoader env.Loader) (*Conf, error) {
+	var envOpts = make([]env.ParseOption, 0)
+	if envLoader != nil {
+		envOpts = append(envOpts, env.WithLoader(envLoader))
+	}
+
+	esURL, err := env.GetOrDefault(
+		"ELASTICSEARCH_URL",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for ELASTICSEARCH_URL: %w", err)
+	}
+
+	index, err := env.GetOrDefault(
+		"ELASTICSEARCH_INDEX",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for ELASTICSEARCH_INDEX: %w", err)
+	}
+
+	apiKey, err := env.GetOrDefault(
+		"ELASTICSEARCH_API_KEY",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for ELASTICSEARCH_API_KEY: %w", err)
+	}
+
+	switch {
+	case esURL == "":
+		return nil, errors.Errorf("variable 'ELASTICSEARCH_URL' needs to be set")
+	case index == "":
+		return nil, errors.Errorf("variable 'ELASTICSEARCH_INDEX' needs to be set")
+	case apiKey == "":
+		return nil, errors.Errorf("variable 'ELASTICSEARCH_API_KEY' needs to be set")
+	}
+
+	return &Conf{
+		ElasticsearchURL:    esURL,
+		ElasticsearchIndex:  index,
+		ElasticsearchApiKey: apiKey,
+	}, nil
+}
+
+// Report logs the findings in json format in the target elastcisearch.
+func (e esLogger) Report(
+	ctx context.Context,
+	findings []*vf.VulnerabilityFinding,
+) error {
+	logger := component.
+		LoggerFromContext(ctx).
+		With(slog.Int("num_findings", len(findings)))
+	client, err := e.getESClient()
+	if err != nil {
+		return errors.Errorf("could not connect to elasticsearch err: %w", err)
+	}
+	for _, finding := range findings {
+		b, err := protojson.Marshal(finding.Finding)
+		if err != nil {
+			return errors.Errorf("could not json marshal finding: %w", err)
+		}
+		client.Index(e.Conf.ElasticsearchIndex, bytes.NewBuffer(b))
+		logger.Info("found finding", slog.String("finding", string(b)))
+	}
+
+	return nil
+}
+
+func dumpStringResponse(res *esapi.Response) string {
+	return res.String()
+}
+
+func (e esLogger) getESClient() (*elasticsearch.Client, error) {
+	var es *elasticsearch.Client
+	var err error
+	type esInfo struct {
+		Version struct {
+			Number string `json:"number"`
+		} `json:"version"`
+	}
+
+	es, err = elasticsearch.NewClient(elasticsearch.Config{
```

Comment:
Can we initialise this in the main and fail fast + inject it in the reporter struct? It would be tidier and easier to test